subprocess/subtree_oom_darwin.go

```diff
@@ -0,0 +1,82 @@
+// +build darwin
+
+package subprocess
+
+import (
+	"bufio"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func (o *OOMTracker) Clear() error {
+	var err error
+	o.IsSudo, err = isSudo()
+	if err != nil {
+		return errors.Wrap(err, "error checking sudo")
+	}
+
+	if o.IsSudo {
+		return errors.Wrap(exec.Command("sudo", "log", "erase", "--all").Run(), "error clearing log")
+	}
+
+	return errors.Wrap(exec.Command("log", "erase", "--all").Run(), "error clearing log")
+}
+
+func (o *OOMTracker) Check() error {
+	wasOOMKilled, pids, err := analyzeLogs(o.IsSudo)
+	if err != nil {
+		return errors.Wrap(err, "error searching log")
+	}
+	o.WasOOMKilled = wasOOMKilled
+	o.Pids = pids
+	return nil
+}
+
+func analyzeLogs(isSudo bool) (bool, []int, error) {
+	var cmd *exec.Cmd
+	wasOOMKilled := false
+	pidChannel := make(chan *int)
+
+	if isSudo {
+		cmd = exec.Command("sudo", "log show")
+	} else {
+		cmd = exec.Command("log show")
+	}
+	cmdReader, err := cmd.StdoutPipe()
+	if err != nil {
+		return false, []int{}, errors.Wrap(err, "error creating StdoutPipe for log command")
+	}
+	scanner := bufio.NewScanner(cmdReader)
+	go func() {
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.Contains(line, "swap") { // OOM killed!
+				wasOOMKilled = true
+				pid := getPidFromLog(line)
+				if pid != nil {
+					pidChannel <- pid
+				}
+			}
+		}
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		return false, []int{}, errors.Wrap(err, "Error starting log command")
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		return false, []int{}, errors.Wrap(err, "Error waiting on log command")
+	}
+
+	close(pidChannel)
+	pids := []int{}
+	for pid := range pidChannel {
```

Comment:
This feels very much like a `range` and pointer kind of bug (e.g, https://play.golang.org/p/75PfVipqzMx). It might be better to instead return a bool (indicating found a pid or not) and an integer from the `getPidFromLog` function instead of a pointer to an int, since removing pointers fixes that bug (e.g., https://play.golang.org/p/rgA3d0N0rEF).

Reply:
Actually, I'm totally wrong. The bug is actually in the sender, not the receiver (https://play.golang.org/). Ignore this.